### DISCUSSION TITLE
How to enable the new default EnableObjectFiltering?

### DISCUSSION THREAD

**Discussion by lwcorp** [sandboxie-plus/Sandboxie#1769] [Q&A Feedback]
In [v1.0.16 / 5.55.16](https://github.com/sandboxie-plus/Sandboxie/releases/tag/1.0.16) "**EnableObjectFiltering** option is now on by default, although this only applies for new installations, hence its recommend for existing installation to go to **settings->advanced** and enable it explicitly."

However, there doesn't seem to be anything that is called either **Settings** or **Advanced**.

Does it actually mean **Options=>Global Settings=>Advanced Config**? But if so, the only **Filtering** option there is network access, which is not at all related to "sandboxie will not allow for PROCESS_VM_READ on unsandboxed processes or processes belonging to other boxes."
![image](https://user-images.githubusercontent.com/1773306/161617518-2d75b5ff-7bd1-44d6-aa89-90f2007633e2.png)

Alternatively, do "new installations" mean "new installations of actual sandboxes? I.e. does it mean **Sandbox Options=>Advanced Options=>Isolation**? If so, it doesn't seem to have anything that even contains the words **Object** or **Filtering**.

I've also created a new sandbox, but it had exactly the same settings in that screen.

Can you post a clear screenshot of how to enable **EnableObjectFiltering** in the UI?

![image](https://user-images.githubusercontent.com/1773306/161615939-dc49d440-6446-458a-b71c-500e59a060ad.png)
![image](https://user-images.githubusercontent.com/1773306/161616077-97e531f7-26e4-414f-9e22-84e9574926ae.png)

**Comment by DavidXanatos** [comment]
Its in the global options:

![grafik](https://user-images.githubusercontent.com/3890945/161617882-3e7bd5b8-71c0-4e5d-bfd3-5cd2ce8538c7.png)

**Comment by lwcorp** [comment]
Thanks! So is it about **network access**, **not** about "sandboxie will not allow for PROCESS_VM_READ on unsandboxed processes or processes belonging to other boxes"?

**Reply by DavidXanatos** [reply]
The only feature there about network is Windows Filtering PLatform
the other options are other things.
what you need os Activate Kernel Mode Object Fitlering

**Reply by lwcorp** [reply]
But as you can see in my previous screenshot **Activate Kernel Mode Object Filtering** was already activated in my existing installation. Is it because of using the portable version?

P.S.
Should "Windows Filtering PLatform" be on? It's only off because that must have been the default.

**Comment by bjm234** [comment]
@lwcorp 
FWIW ~ as test: clean install Plus 1.0.17
1) Uninstall 1.0.15 + Remove configuration files 
2) clean install Plus 1.0.17
my Sandboxie Plus 1.0.17 clean install has two Global Settings boxes checked by default that were not checked with my prior new build over-installs.
<img width="704" alt="png_14434" src="https://user-images.githubusercontent.com/25015553/162237523-10e67bdf-9e11-4cb5-821b-e870659fa734.png">

> System call hooking for Win32k system calls is now enabled by default, it is still used only for a hand full of calls currently, as required to get chromium Hardware Acceleration acceleration to work properly.

https://github.com/sandboxie-plus/Sandboxie/releases/tag/1.0.4

**Reply by lwcorp** [reply]
If you see my original screenshot, it has the very same checkboxes even though it's not a clean install, which is my I started this discussion.
It was advertised as a "clean install-only" option and seemingly it's not.

**Reply by bjm234** [reply]
@lwcorp 
The first time ... I noticed the two Global Settings enabled by default is after my 1.0.17 "clean install".  My 1.0.15 did not have the two Global Settings enabled.  Previously, for some time.  All my new build installs were over-installs.  I've been using same Sandboxie.ini for long time. 
After reading this Topic.  I got curious.

**Reply by DavidXanatos** [reply]
@lwcorp look it depands on the state of your previous install, 
1. you could have enabled that earlier manually alleady
2. if you never changed any settings there, there would be no entry in the sandboxie ini and it would therefor take the new default.

Only if you have an existing install and changed settings that caused the n value to be written to the ini it would stay off, since i assumed that would be the majority case i wanted to bring the users attention to that and ensure thay would enable it.

**Reply by lwcorp** [reply]
>     2. if you never changed any settings there, there would be no entry in the sandboxie ini and it would therefor take the new default.
Oh, so that's what caused the confusion. The new version's statement caused to think it's expected to still be off for upgrades. You might want to edit it and mention "if you previously turned it off, it'll still be off after the upgrade. Otherwise it'll get enabled after the upgrade".
